src/main/java/io/xcherry/core/client/BasicClient.java

```diff
@@ -124,6 +128,40 @@ public <T> T invokeRPC(final ProcessExecutionRpcRequest request, final Class<T>
         return clientOptions.getObjectEncoder().decodeFromEncodedObject(response.getOutput(), returnType);
     }
 
+    /**
+     * Wait for a process execution to complete within the specified timeout seconds.
+     *  1. If the process execution is still running after the timeout seconds, the response will be returned with the timeout field being true.
+     *  2. If the process execution has stopped by system (e.g., due to re-balancing after increasing/decreasing the shards in the server), the response will be returned with the stopBySystem field being true.
+     *  3. In other cases, the current status of the process execution will be returned.
+     *
+     * @param namespace         the namespace in which the operation should be performed.
+     * @param processId         a unique identifier used to differentiate between different executions of the same process type.
+     * @param timeoutInSeconds  a value less than or equal to 30.
+     */
+    public ProcessExecutionWaitForCompletionResponse waitForProcessCompletion(
+        final String namespace,
+        final String processId,
+        final int timeoutInSeconds
+    ) {
+        final int timeout;
+        if (timeoutInSeconds <= 0 || timeoutInSeconds > DEFAULT_WAIT_FOR_TIMEOUT) {
+            timeout = DEFAULT_WAIT_FOR_TIMEOUT;
+        } else {
+            timeout = timeoutInSeconds;
+        }
+
+        final ProcessExecutionWaitForCompletionRequest request = new ProcessExecutionWaitForCompletionRequest()
+            .namespace(namespace)
+            .processId(processId)
+            .timeoutSeconds(timeout);
```

Comment:
This timeout may not work unless you set the timeout for Feign client. 

This is a bit complicated but there are :
connectTimeout
readTimeout
writeTimeout

https://blog.cloudflare.com/the-complete-guide-to-golang-net-http-timeouts

https://javadoc.io/doc/io.github.openfeign/feign-core/10.9/feign/Request.Options.html

Reply:
Do you mean the default timeout in feign Request.Options can be less than the specified value here? Maybe we can set the tiemout in feign to be 30s by default to avoid it.